### PR TITLE
Make error responses better

### PR DESCRIPTION
**Pull request by codehackerr**

### CODE REVIEW

**Comment by codehackerr** [comment]
This is a DRAFT PR for enhancing the understanding.

Decompressed HTTP responses + error ( GenericSwaggerError) object is holding byte[]. This is cryptic to read.
Here I changed one instance to string for demo purpose. We can replace the old Swagger error object, V2 may not be required.

Second request, I believe is to expose context.Context to executor, workflow etc . Currently it's already a parameter to clients. Client's description is confusing. https://orkesinternal.slack.com/archives/D075LNWQGFP/p1729627367176919

**Comment by shaileshpadave** [comment]
IMO returning response in byte[] is more useful for following reasons : 

1. Flexibility in Handling Various Response Types: 
By storing the body as []byte, the API client allows for flexibility in handling various response content types. The content of an error response can vary (JSON, XML, plain text, binary, etc.), and using a byte array makes it easy to process these formats as needed.

2. Performance Considerations: 
Using a []byte instead of directly converting to a string or specific type upfront can be more efficient. Since the client may not know the format of the error response ahead of time, keeping the raw bytes gives you more control over when and how you parse it.

Also by using raw `[]byte` for the body, the generated client can remain as generic as possible and support different response types or error formats without the need for specialised error types for every API response.

**Comment by codehackerr** [comment]
Few things.
- We have hard coded application/json for all resources ( so binary does not apply in general )
- Error body should be textual content ( json, plain .. whatever) .
```newErr := GenericSwaggerErrorV2{
			body:  string(responseBody),
			error: httpResponse.Status,
		} 
```
- we don't have to change the response representation. It can still be byte[]

File: sdk/client/generic_swagger_error.go [hunk 1]
```diff
@@ -18,16 +18,37 @@ type GenericSwaggerError struct {
...
 
+type GenericSwaggerErrorV2 struct {
+	body  string
+	error string
+	model interface{}
+}
+
 // Error returns non-empty string if there was an error.
-func (e GenericSwaggerError) Error() string {
+func (e GenericSwaggerErrorV2) Error() string {
```
Comment (codehackerr) [inline sdk/client/generic_swagger_error.go:28]: Won't need `GenericSwaggerErrorV2` , just for demo

**Review by jmigueprieto — CHANGES_REQUESTED** [review]
@codehackerr can you please add a description of what we want to achieve with this PR?

Based on what I know about this, what we want to fix is that our Client/SDK is suppressing the HTTP status code. That causes issues while debugging. How is this addressing it?

**Comment by codehackerr** [comment]
The changes in this PR are based on a wrong interpretation. Closing the PR.

**Pull request closed without merging**